tart2ms/tart2ms.py

```diff
@@ -596,51 +674,64 @@ def __twelveball(direction):
                             f"{np.ceil(obs_length / (rayleigh_crit / sidereal_rate) * 100):.0f}% "
                             f"of the instrument resolution during the course of this observation")
 
+        if fill_model:
+            if not AFRICANUS_DFT_AVAIL:
+                raise RuntimeError("Cannot predict model visibilities. Please install codex-africanus package")
+            if not sources:
+                LOGGER.critical("You have requested to predict a model for GNSS sources, however one or more of "
+                                "the databases you've provided contains no GNSS source information. The MODEL_DATA "
+                                "column of your database may be incomplete")
+            else:
+                if len(epoch_s) != len(sources):
+                    raise RuntimeError(
+                        "If sources are specified then we expected epochs to be of same size as sources list")
+                model_data = da.zeros_like(dask_data)
+                spwi_chan_freqs = da.from_array(spw_chan_freqs[spw_i])
+                for dataset_i, (epoch_s_i, sources_i) in enumerate(zip(epoch_s, sources)):
+                    # get J2000 RADEC
+                    sources_radec = np.empty((len(sources_i), 2))
+                    for src_i, src in enumerate(sources_i):
+                        name = src['name']
+                        # Convert to J2000
+                        dir_altaz = SkyCoord(alt=src['el']*u.deg, az=src['az']*u.deg, obstime=obstime[0],
```

Comment:
We should add helper functions for conversion from altaz -> radec, particularly as I would like to move to a JSON formatted sky model being provided on the command line. So perhaps we write something to parse the sky model into a list of  SkyCoord objects.  Perhaps can discuss offline